flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -97,24 +123,26 @@ public void checkpointState(
 		// Step (3): Take the state snapshot. This should be largely asynchronous, to not
 		//           impact progress of the streaming topology
 
-		CheckpointStreamFactory storage = checkpointStorage.resolveCheckpointStorageLocation(
-			checkpointMetaData.getCheckpointId(),
-			checkpointOptions.getTargetLocation());
-
 		long startSyncPartNano = System.nanoTime();
 
 		HashMap<OperatorID, OperatorSnapshotFutures> operatorSnapshotsInProgress = new HashMap<>(operatorChain.getNumberOfOperators());
+		ChannelStateWriteResult channelStateWriteResult =
```

Comment:
should we add some `TODO` before `getWriteResult`, otherwise it seems hard to understand from this commit that we actually have not written anything before get the result.

Reply:
There is todo in `StreamTask` near `SubtaskCheckpointCoordinatorImpl` constructor:
```
false); // todo: pass true if unaligned checkpoints enabled
```

(I expect the exact type of this option could change)